From bdb63eb5413c5b44dac08bf42738a09e252fd4ef Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Micha=C5=82=20Gapi=C5=84ski?= <mike@gapinski.eu>
Date: Thu, 29 Jun 2023 17:42:36 +0000
Subject: [PATCH 1/1] Remove setup wizard

---
 arm64/Android.bp | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/arm64/Android.bp b/arm64/Android.bp
index 1ea3768..a37eb3c 100644
--- a/arm64/Android.bp
+++ b/arm64/Android.bp
@@ -74,16 +74,16 @@ android_app_import {
 	product_specific: true,
 }
 
-android_app_import {
-	name: "SetupWizard",
-	owner: "gapps",
-	apk: "proprietary/system_ext/priv-app/SetupWizard/SetupWizard.apk",
-	overrides: ["Provision"],
-	presigned: true,
-	dex_preopt: {
-		enabled: false,
-	},
-	privileged: true,
-	system_ext_specific: true,
-}
+//android_app_import {
+//	name: "SetupWizard",
+//	owner: "gapps",
+//	apk: "proprietary/system_ext/priv-app/SetupWizard/SetupWizard.apk",
+//	overrides: ["Provision"],
+//	presigned: true,
+//	dex_preopt: {
+//		enabled: false,
+//	},
+//	privileged: true,
+//	system_ext_specific: true,
+//}
 
-- 
2.34.1

