From a6e7a244ddd84677b33f7cbcd461a3b70fe9cf93 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Tue, 4 Apr 2023 21:48:18 +0300
Subject: [PATCH 4/6] GLODROID: Speed-up boot

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 configs/rpi_4_defconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/configs/rpi_4_defconfig b/configs/rpi_4_defconfig
index f03b56ad6a..4ab246315a 100644
--- a/configs/rpi_4_defconfig
+++ b/configs/rpi_4_defconfig
@@ -65,3 +65,4 @@ CONFIG_VIDEO_BCM2835=y
 CONFIG_CONSOLE_SCROLL_LINES=10
 CONFIG_PHYS_TO_BUS=y
 CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_BOOTDELAY=1
-- 
2.37.2

