From ba29c6d7ac8fe808013821a55c61ccdbe2937a94 Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Sat, 25 Nov 2023 08:25:11 +0200
Subject: [PATCH 11/11] GLODROID: Reduce boot delay countdown timer from 1S to
 100uS

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 common/autoboot.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/common/autoboot.c b/common/autoboot.c
index 5d331991c1..92f91d7796 100644
--- a/common/autoboot.c
+++ b/common/autoboot.c
@@ -385,7 +385,7 @@ static int abortboot_single_key(int bootdelay)
 
 	while ((bootdelay > 0) && (!abort)) {
 		--bootdelay;
-		/* delay 1000 ms */
+		/* delay 100 ms */
 		ts = get_timer(0);
 		do {
 			if (tstc()) {	/* we got a key press	*/
@@ -399,7 +399,7 @@ static int abortboot_single_key(int bootdelay)
 				break;
 			}
 			udelay(10000);
-		} while (!abort && get_timer(ts) < 1000);
+		} while (!abort && get_timer(ts) < 100);
 
 		printf("\b\b\b%2d ", bootdelay);
 	}
-- 
2.39.2

