From 08bf277bb68ef649949a83eb908a35666dc7fccf Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Fri, 6 Oct 2023 22:20:52 +0300
Subject: [PATCH 08/12] base: Remove deprecated BOARD_BUILD_SYSTEM_ROOT_IMAGE
 definition

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 common/base/board.mk | 1 -
 1 file changed, 1 deletion(-)

diff --git a/common/base/board.mk b/common/base/board.mk
index 82c4292..73158c5 100644
--- a/common/base/board.mk
+++ b/common/base/board.mk
@@ -97,7 +97,6 @@ TARGET_COPY_OUT_VENDOR_DLKM := vendor_dlkm
 
 # Root image
 TARGET_COPY_OUT_ROOT := root
-BOARD_BUILD_SYSTEM_ROOT_IMAGE := false
 
 BOARD_EXT4_SHARE_DUP_BLOCKS := true
 
-- 
2.39.2

