From a9099ba4059591b0246f86031f8483129a2a97db Mon Sep 17 00:00:00 2001
From: Roman Stratiienko <r.stratiienko@gmail.com>
Date: Sat, 8 Apr 2023 02:00:59 +0300
Subject: [PATCH 10/11] codecs: Build ffmpeg project using aospext

Signed-off-by: Roman Stratiienko <r.stratiienko@gmail.com>
---
 common/codecs/board.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/common/codecs/board.mk b/common/codecs/board.mk
index fa45899..9358763 100644
--- a/common/codecs/board.mk
+++ b/common/codecs/board.mk
@@ -6,3 +6,6 @@
 
 BOARD_BUILD_AOSPEXT_DAV1D := true
 BOARD_DAV1D_SRC_DIR := glodroid/vendor/dav1d
+
+BOARD_BUILD_AOSPEXT_FFMPEG := true
+BOARD_FFMPEG_SRC_DIR := glodroid/vendor/ffmpeg
-- 
2.37.2

