From 423770562d1c37cb15a01574782773e738d003df Mon Sep 17 00:00:00 2001
From: Konsta <konsta09@gmail.com>
Date: Sun, 6 Apr 2025 16:44:43 +0300
Subject: [PATCH 40/40] android: fix including drm header

---
 libavutil/hwcontext_drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libavutil/hwcontext_drm.c b/libavutil/hwcontext_drm.c
index dac9befae9..c8c3fe5cf4 100644
--- a/libavutil/hwcontext_drm.c
+++ b/libavutil/hwcontext_drm.c
@@ -32,7 +32,7 @@
 #endif
 
 #include <drm.h>
-#include <libdrm/drm_fourcc.h>
+#include <drm_fourcc.h>
 #include <xf86drm.h>
 
 #include "avassert.h"
-- 
2.34.1

